src/main/java/demon/Commands.java

```diff
@@ -0,0 +1,285 @@
+package demon;
+
+import java.io.IOException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * This class serves as a central point for managing and executing a variety of user commands.
+ * It encapsulates the logic for each command as individual methods, allowing for easy execution
+ * based on user input. Commands such as 'list', 'unmark', 'mark', 'delete', and others are supported, each
+ * corresponding to a specific method within this class.
+ */
+public class Commands {
+    Storage storage;
+
+    public Commands(String filePath) {
+        this.storage = new Storage(filePath);
+    }
+    /**
+     * Prints out all tasks saved numbered in sequence.
+     * No tasks will be printed if there is no saved tasks.
+     *
+     * @param tasks an array containing all tasks.
+     */
+    public void list(TaskList tasks) {
```

Comment:
I like how you used 'tasks' to show that TaskList contains multiple tasks 👍

Reply:
Thank you 🙏